doc/guides/porting/arch.rst

```diff
@@ -795,7 +795,7 @@ details:
 
 Some architectures may need to update software memory management structures
 or modify hardware registers on another CPU when memory domain APIs are invoked.
-If so, CONFIG_ARCH_MEM_DOMAIN_SYNCHRONOUS_API must be selected by the
+If so, :option:`CONFIG_ARCH_MEM_DOMAIN_SYNCHRONOUS_API` must be selected by the
```

Comment:
nit: use `:kconfig:` to reference Kconfig options

Reply:
done.  Thanks.